pkg/apis/componentconfig/v1alpha1/zz_generated.conversion.go

```diff
@@ -814,6 +814,7 @@ func autoConvert_v1alpha1_LeaderElectionConfiguration_To_componentconfig_LeaderE
 	out.LeaseDuration = in.LeaseDuration
 	out.RenewDeadline = in.RenewDeadline
 	out.RetryPeriod = in.RetryPeriod
+	out.LockType = in.LockType
```

Comment:
I would prefer `LockResource`.  It also needs to be a group and resource, not just one or the other.  You probably don't want to perform discovery, so I'd recommend doing `LockResource` and `LockResourceGroup` or similar.